### ISSUE TITLE
Azure Let's Encrypt is not being renewed

### ISSUE THREAD

**Issue by maxkonyukh** [sjkp/letsencrypt-siteextension#166]
Have this sort of exception:
**System.Threading.Tasks.TaskCanceledException**: System.Threading.Tasks.TaskCanceledException: A task was canceled.
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 at LetsEncrypt.SiteExtension.Core.Services.BaseAuthorizationChannelgeProvider.<Authorize>d__5.MoveNext() in J:\Projects\letsencrypt-siteextension\LetsEncrypt.SiteExtension.Core\Services\BaseAuthorizationChannelgeProvider.cs:line 86
 --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
 at LetsEncrypt.SiteExtension.Core.Services.BaseAuthorizationChannelgeProvider.<Authorize>d__5.MoveNext() in J:\Projects\letsencrypt-siteextension\LetsEncrypt.SiteExtension.Core\Services\BaseAuthorizationChannelgeProvider.cs:line 131
 --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 at LetsEncrypt.SiteExtension.Core.Services.AcmeService.<RequestCertificate>d__5.MoveNext() in J:\Projects\letsencrypt-siteextension\LetsEncrypt.SiteExtension.Core\Services\AcmeService.cs:line 39
 --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 at LetsEncrypt.SiteExtension.Core.CertificateManager.<RequestAndInstallInternalAsync>d__10.MoveNext() in J:\Projects\letsencrypt-siteextension\LetsEncrypt.SiteExtension.Core\CertificateManager.cs:line 150
 --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 at LetsEncrypt.SiteExtension.Core.CertificateManager.<RenewCertificate>d__8.MoveNext() in J:\Projects\letsencrypt-siteextension\LetsEncrypt.SiteExtension.Core\CertificateManager.cs:line 117
 --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 at LetsEncrypt.SiteExtension.Functions.<RenewCertificate>d__1.MoveNext() in J:\Projects\letsencrypt-siteextension\LetsEncrypt.SiteExtension.WebJob\Functions.cs:line 61
 --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 at Microsoft.Azure.WebJobs.Host.Executors.FunctionInvoker`1.<InvokeAsync>d__0.MoveNext()
 --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithWatchersAsync>d__31.MoveNext()
 --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithLoggingAsync>d__2c.MoveNext()
 --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithLoggingAsync>d__13.MoveNext()
 --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
 at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithLoggingAsync>d__13.MoveNext()
 --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<TryExecuteAsync>d__1.MoveNext()

**Comment by fredrik-annytab** [comment]
I also have that exception from time to time, I updated to the latest version today (0.7.5). I have a lot of domain names on the same site, 33 certificates.

**Comment by maxkonyukh** [comment]
So did upgrade resolve that exception?

**Comment by sjkp** [comment]
This error happens when the extension is unable to request the challenge file success. After the extension has written the challenge file to disk, it tries to request it the same way that letsencrypt would, and for some reason your site fails to respond to that request. 

I will add some try catch to that part of the code, so that it will retry even if the request completely fails. But I would encurage you to check if you can browse the challenge file yourself, from the .well-known/acme-challenge/ folder.

**Comment by fredrik-annytab** [comment]
I have three RenewCertificate functions, I did a run of Functions.RenewCertificate | 172 0 and it was a success. The first two functions have a orange warning: Host not running.

FUNCTION NAME | STATISTICS
Functions.RenewCertificate | 441 8
Functions.SetupHostNameAndCertificate | 181 1
Functions.AddCertificate | 47 2
Functions.Cleanup | 172 1
Functions.RenewCertificate | 172 0
Functions.AddCertificate | 45 4
Functions.Cleanup | 175 2
Functions.RenewCertificate | 168 12

I have 130 files in the .well-known/acme-challenge/ folder and 1 web.config file. I tested to reach the first challenge file (Created: 2017-09-13) and it was possible to browse to it.
